### DISCUSSION TITLE
Google Analytics implementation

### DISCUSSION THREAD

**Discussion by zurtox** [dotnet/maui#13339] [Q&A]
### Description

How is the process of making the implementation of Google Analytics in .Net MAUI?
We have working at implementing Firebase Analytics in our MAUI application, unfortunately the process has been a real pain about the target frameworks of .Net 6.0 and/or 7.0.
Does it has to be implemented like Xamarin Forms or it will be a package which contains support for all the platforms in one?

### Steps to Reproduce

1. Create a .Net MAUI App
2. Add Firebase Analytics Nuget package (but which one?)

### Link to public reproduction project repository

-

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

iOS and Android

### Did you find any workaround?

Maybe implementing as Xamarin? Or stuck in Xamarin until the light gets on in any future oficial Nuget library for MAUI

**Comment by Xyncgas** [comment]
GA always wants you to talk to them, so either find a library that does that, or use a MAUI.blazor project and you can do it in html like you always did

**Comment by Ghostbird** [comment]
I feel that you should _not_ have labeled this as a _bug_. The bug label clearly states that it is for when _something_ isn't working. This is a feature request. The _something_ doesn't exist yet.

Furthermore such a feature will never be part of MAUI, for reasons which ought to be obvious. As such, you should rather look around which libraries already offer MAUI support. If none do, I think you have four options.

- Don't use MAUI
- Use MAUI but not Google Analytics
- Help an Open Source library out and make them compatible with MAUI. You can see this is happening at [Microcharts](https://github.com/microcharts-dotnet/Microcharts/pull/325)
- Create your own library (either open or closed source)

The same applies to your other question about Google Ads.

**Comment by goenning** [comment]
I'm working on a privacy-friendly analytics alternative to Google in case you're interested. It's solely focused on apps, with support for MAUI: https://aptabase.com/for-maui

You can either self host or use the cloud version https://github.com/aptabase/aptabase 

fun fact: the backend is also in C# :)

**Comment by nll** [comment]
We have the most recent Xamarin.Firebase packages that do target .NET6+ that can be use
- Android - [Xamarin.Firebase.Analytics](https://www.nuget.org/packages/Xamarin.Firebase.Analytics)
- iOS - [Xamarin.Firebase.iOS.Analytics](https://www.nuget.org/packages/Xamarin.Firebase.iOS.Analytics)

So I would say that it should be implemented like it was for Xamarim.Forms

**Comment by mhrastegari** [comment]
We're using these versions for Android if it helps

```
    <PackageReference Include="Plugin.LocalNotification" Version="9.2.0"></PackageReference>
    <PackageReference Include="Xamarin.Build.Download" Version="0.11.3"></PackageReference>
    <PackageReference Include="Xamarin.Firebase.Analytics" Version="119.0.1.2"></PackageReference>
    <PackageReference Include="Xamarin.Firebase.Crashlytics" Version="118.2.1.2"></PackageReference>
    <PackageReference Include="Xamarin.Firebase.Messaging" Version="122.0.0.2"></PackageReference>
    <PackageReference Include="Xamarin.Google.Dagger" Version="2.40.5"></PackageReference>
    <PackageReference Include="Xamarin.GooglePlayServices.Base" Version="118.0.1"></PackageReference>
```
